### PR TITLE
Add packages attribute to lockfile object

### PR DESCRIPTION
**Pull request by athos-ribeiro**
This is a list with the new Package instances for each package
represented in the parsed lockfile.

Signed-off-by: Athos Ribeiro <athos@redhat.com>

### CODE REVIEW

**Comment by athos-ribeiro** [comment]
@nirzari

File: tests/test_lockfile.py [hunk 1]
```diff
@@ -91,3 +91,53 @@ def test_unknown_version(data):
...
+    url = 'https://example.com/breakfast/1.1.1.tar.gz'
+    data = f'breakfast@^1.1.1:\n  version "2.0.0"\n  resolved "{url}"\n  integrity someHash'
+    lock = lockfile.Lockfile.from_str(data)
+    packages = lock.packages()
+    assert len(packages) == 1
+    assert packages[0].name == 'breakfast'
+    assert packages[0].version == '2.0.0'
+    assert packages[0].checksum == 'someHash'
+    assert packages[0].url == url
+    assert packages[0].relpath is None
```
Comment (nirzari) [inline tests/test_lockfile.py:143]: Should we test a scenario where relpath is not None?

Comment (athos-ribeiro) [reply tests/test_lockfile.py:143]: Done

**Comment by chmeliik** [comment]
Should this have special handling for the case when there are multiple packages on one line? e.g.
```
foo, bar
  version "1.0.0"
```
Or is this something that we want to handle on the parser level? https://github.com/containerbuildsystem/pyarn/blob/9db64f4d6ff77f28db699b0258c52b89743243ca/pyarn/parser.py#L42

(or is it irrelevant?)

**Comment by athos-ribeiro** [comment]
> Should this have special handling for the case when there are multiple packages on one line? e.g.
> 
> ```
> foo, bar
>   version "1.0.0"
> ```

these would actually be `foo@ver1, foo@ver2, ... foo@verN`. The regular expression is handling this by fetching the first occurrence of `foo`.

> 
> Or is this something that we want to handle on the parser level?
> 
> https://github.com/containerbuildsystem/pyarn/blob/9db64f4d6ff77f28db699b0258c52b89743243ca/pyarn/parser.py#L42
> 
> (or is it irrelevant?)

I believe this is an open question for now. I am wondering if we have any use for the resolved values for each specific version of a package at all. From a Cachito perspective, it shouldn't matter unless different versions of a package would be resolved from an external source (VCS, HTTP) and from the Yarn registry. It should still be possible to map those in Cachito by looking at pyarn's Lockfile.data dict.

**Comment by chmeliik** [comment]
> these would actually be `foo@ver1, foo@ver2, ... foo@verN`. The regular expression is handling this by fetching the first occurrence of `foo`.

Oh I see, that seems reasonable. Should there be a unit test for this?

**Comment by athos-ribeiro** [comment]
> > these would actually be `foo@ver1, foo@ver2, ... foo@verN`. The regular expression is handling this by fetching the first occurrence of `foo`.
> 
> Oh I see, that seems reasonable. Should there be a unit test for this?

done!

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Pull request merged**